qt/DetailsDialog.cc

```diff
@@ -386,6 +389,19 @@ void DetailsDialog::onSessionCalled(Session::Tag tag)
     }
 }
 
+void DetailsDialog::onButtonBoxClicked(QAbstractButton* button)
+{
+    if (ui_.dialogButtons->standardButton(button) == QDialogButtonBox::Close)
+    {
+        auto labels_text = ui_.labelsTextEdit->toPlainText().trimmed();
+        QString re = QStringLiteral("((,|;)\\s*)");
+        QStringList labels_list = labels_text.split(QRegularExpression(re), Qt::SkipEmptyParts);
+
+        torrentSet(TR_KEY_labels, labels_list);
```

Comment:
> This unconditionally calls `torrentSet()` every time the details dialog is closed, even if the labels haven't changed. This is unnecessary overhead; we should check to see if the result differs from the original first. 😸 See also my other suggestion about `Torrent.labels()` returning a `QStringList` since that would make comparison here nice & easy
> 
> Another issue -- if there's >1 torrent being shown, then `ui.labelsTextEdit` will be empty and so closing the dialog will delete their labels 💥

done

> Maybe we should update when the edit field loses focus iff different from before, instead of when the button box is clicked

This behavior is very unintuitive and can sometimes be frustrating imo. As a user, I would expect that settings only change when I express clear intent, i.e. pressing a button. The way it currently works also enables users to just discard any unwanted edits by clicking the X button to close the window.